kolibri/core/decorators.py

```diff
@@ -320,3 +321,26 @@ def wrapped_view(*args, **kwargs):
 
     wrapped_view.signin_redirect_exempt = True
     return wraps(view_func)(wrapped_view)
+
+
+def cache_no_user_data(view_func):
+    """
+    Set appropriate Vary on headers on a view that specify there is
+    no user specific data being rendered in the view.
+    In order to ensure that the correct Vary headers are set,
+    the session is deleted from the request, as otherwise Vary cookies
+    will always be set by the Django session middleware.
+    This should not be used on any view that bootstraps user specific
+    data into it - this will remove the headers that will make this vary
+    on a per user basis.
+    """
+
+    def inner_func(*args, **kwargs):
+        request = args[0]
+        del request.session
+        response = view_func(*args, **kwargs)
+        patch_response_headers(response, cache_timeout=300)
+        response["Vary"] = "accept-encoding, accept"
```

Comment:
I believe so, not varying on these could cause problems if we sent the wrong mimetype back to a request, or tried to compress for a client that couldn't handle compression.

`accept-charset` is another possibility, but I have never seen our backend add `accept-charset` to the Vary header. My hope is this would be subsumed by language?